supercloud.md

```diff
@@ -0,0 +1,78 @@
+# Running Experiments on Supercloud
+
+## First Time Setup
+
+### Getting Access to Supercloud
+
+* If you do not yet have a supercloud account, you will need to [request one](https://supercloud.mit.edu/requesting-account). 
+* After receiving an account, follow the instructions to ssh into supercloud.
+* Once logged in to supercloud, [create an ssh key and add it to your github.com account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
+* The most important rule about using supercloud is that you should not run any intensive jobs on a login node. Instead, launch batch jobs, or use an interactive session. Both options are illustrated below.
+
+### Installing this Repository
+
+Run the following commands to install this repository.
+
+```
+# Set `PYTHONHASHSEED` to 0 by default (see our README.md for explanation).
+echo "export PYTHONHASHSEED=0" >> ~/.bashrc
+# Clone the repository.
+git clone git@github.com:Learning-and-Intelligent-Systems/predicators.git
+# Set up conda with Python 3.9.
+module unload anaconda
+module load anaconda/2021b
+conda create --name predicators python=3.9
+conda init bash  # will need to restart shell after this
+conda activate predicators
+# Install the predicators dependencies.
+cd predicators
+mkdir /state/partition1/user/$USER
+export TMPDIR=/state/partition1/user/$USER
+pip install -r requirements.txt
+# Add a shortcut for activating the conda env and switching to this repository.
+echo -e "predicate() {\n    cd ~/predicators\n    conda activate predicators\n}" >> ~/.bashrc
+```
+To test if it worked:
+```
+# Start an interactive session.
+LLsub -i
+# Activate conda and switch to the repository.
+predicate
+# Run a short experiment.
+python src/main.py --env cover --approach oracle --seed 0
+# Exit the interactive session.
+exit
+```
+Note that supercloud sometimes hangs, so the experiment may take a while to get started. But once it does, you should see 50/50 tasks solved, and the script should terminate in roughly 2 seconds (as reported at the bottom).
+
+## Running Experiments
+
+To get started, activate the conda environment and switch to the repository. If you followed the instructions above, you can do both with `predicate`.
+
+Before running any experiments, it is good practice to make sure that you have a clean workspace:
+* Make sure that you have already backed up any old results that you want to keep.
+* Remove all previous results: `rm -rf results/* supercloud_* saved_*`.
```

Comment:
I prefer: `rm -f results/* supercloud_logs/* saved_approaches/* saved_datasets/*` so the folders themselves aren't deleted (avoids rare race conditions...)